### PR TITLE
feat: Only show the change cover button and delete button when it's hovering. 

### LINKED ISSUE #2055: [FR] Improve the cover plugin. Only show the change cover button and delete button when it's hovering.
### Description

Screenshot 2023-03-21 at 09 55 08

At the moment, the change cover button and the delete button always show above the cover widget. We want to hide them when the cover isn't hovering.

### Impact

Make the document looks tidier.

### Additional Context

Please refer to:

* `cover_node_widget.dart`

### PR DESCRIPTION
**Pull request by gitstart**
fixes #2055

The change cover button and the delete button always show above the cover widget now they will only show when the mouse is hovering on top of the cover image.

### Issue Video
https://www.loom.com/share/1e9be9d35de946d5a3d787b62bc9edd8

### Fix Video
https://www.loom.com/share/4a59841c6eff48eaaeec1ed60f38bc7e

---
This code was written and reviewed by GitStart Community. Growing future engineers, one PR at a time.

### CODE REVIEW

**Comment by a-wallen** [comment]
This PR needs tests before it can be merged.

**Comment by annieappflowy** [comment]
@gitstart , any progress since the last feedback?

**Comment by gitstart** [comment]
> 

Hello @annieappflowy, 
We are working on the feedback, we will push requested changes soon.

File: frontend/appflowy_flutter/lib/plugins/document/presentation/plugins/cover/cover_node_widget.dart [hunk 1]
```diff
@@ -386,23 +387,31 @@ class _CoverImageState extends State<_CoverImage> {
...
+                      setOverlayButtonsHidden(true);
+                    },
+                    hoverColor: Theme.of(context).colorScheme.surface,
+                    textColor: Theme.of(context).colorScheme.onSurface,
+                    fillColor: Theme.of(context).colorScheme.surface.withOpacity(0.8),
+                    width: 120,
+                    height: 28,
+                    title: LocaleKeys.document_plugins_cover_changeCover.tr(),
+                  )
+                : const SizedBox(),
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/plugins/document/presentation/plugins/cover/cover_node_widget.dart:412]:
```suggestion
                : const SizedBox.shrink(),
```

File: frontend/appflowy_flutter/lib/plugins/document/presentation/plugins/cover/cover_node_widget.dart [hunk 2]
```diff
@@ -386,23 +387,31 @@ class _CoverImageState extends State<_CoverImage> {
         mainAxisSize: MainAxisSize.min,
         children: [
           AppFlowyPopover(
+            onClose: () {
+              isCoverPopoverOpen = false;
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/plugins/document/presentation/plugins/cover/cover_node_widget.dart:391]: Why not set this inside `setOverlayButtonsHidden`?

File: frontend/appflowy_flutter/lib/plugins/document/presentation/plugins/cover/cover_node_widget.dart [hunk 3]
```diff
@@ -411,19 +420,21 @@ class _CoverImageState extends State<_CoverImage> {
...
+                  width: 28,
+                  icon: svgWidget(
+                    'editor/delete',
+                    color: Theme.of(context).colorScheme.onSurface,
+                  ),
+                  onPressed: () {
+                    widget.onCoverChanged(CoverSelectionType.initial, null);
+                  },
+                )
+              : const SizedBox(),
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/plugins/document/presentation/plugins/cover/cover_node_widget.dart:437]:
```suggestion
              : const SizedBox.shrink(),
```

File: frontend/appflowy_flutter/lib/plugins/document/presentation/plugins/cover/cover_node_widget.dart [hunk 4]
```diff
@@ -460,20 +471,28 @@ class _CoverImageState extends State<_CoverImage> {
...
+          maxWidth: screenSize.width,
+          child: Stack(
+            children: [
+              Container(
+                padding: const EdgeInsets.only(bottom: 10),
+                height: double.infinity,
+                width: double.infinity,
+                child: coverImage,
+              ),
+              hasCover ? _buildCoverOverlayButtons(context) : const SizedBox()
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/plugins/document/presentation/plugins/cover/cover_node_widget.dart:493]:
```suggestion
              hasCover ? _buildCoverOverlayButtons(context) : const SizedBox.shrink()
```

**Comment by gitstart** [comment]
Hello @annieappflowy  @Xazin,
All requested changes pushed. You can review it again..
Thank you!

**Comment by LucasXu0** [comment]
Hey, @gitstart. I tested it on your branch and recorded a video. The popover flickers when it pops up.

https://user-images.githubusercontent.com/11863087/232379688-f67c34d5-3836-4717-b10b-4afac29d1960.mov

**Comment by gitstart** [comment]
Thank you for the review, we are working on the changes right now and will push a fix soon.

**Comment by gitstart** [comment]
> Hey, @gitstart. I tested it on your branch and recorded a video. The popover flickers when it pops up.
> 
>  Screen.Recording.2023-04-17.at.12.40.04.mov

Hello @LucasXu0,
Requested changes are pushed ✔.
Feel free to review again.
Thank you🙏

**Pull request merged**